File: core/src/test/scala/diplomacy/variant/standard/VariantSpec.scala
package diplomacy.variant.standard

import diplomacy.UnitSpec
import diplomacy.variant.standard.map._

class VariantSpec extends UnitSpec {
  "The variant" should {
    "define the initial board." in {
      for { power <- variant.initialBoard.map.powers } {
        val numOfSupplys = variant.initialBoard.numberOfSupplyCenters(power)
        val numOfUnits = variant.initialBoard.units count { _.power == power }
        power match {
          case Russia =>
            numOfSupplys should be(4)
            numOfUnits should be(4)
          case _ =>
            numOfSupplys should be(3)
            numOfUnits should be(3)
        }
      }
    }
  }
}

File: test/standard/variant-spec.ts
import * as chai from "chai"
import { standard } from "./../../src/standard"
import { standardRule } from "./../../src/standardRule"
import { standardMap } from "./../../src/standardMap"

const variant = standard.variant
const Utils = standardRule.Utils
const Power = standardMap.Power

const should = chai.should()

describe("Variant", () => {
  it("defines an initial board.", () => {
    const numberOfSupplyCenters = Utils.numberOfSupplyCenters(variant.initialBoard)
    variant.initialBoard.map.powers.forEach(power => {
      const numOfUnits = ([...variant.initialBoard.units].filter(u => u.power === power)).length
      if (power === Power.Russia) {
        should.equal(numberOfSupplyCenters.get(power), 4)
        numOfUnits.should.equal(4)
      } else {
        should.equal(numberOfSupplyCenters.get(power), 3)
        numOfUnits.should.equal(3)
      }
    })
  })
})
